src/Pages/Home/components/AddOperationalCommand.tsx

```diff
@@ -0,0 +1,272 @@
+        name: 'operationalTimes'
+    })
+    const isPermeate = operationalTimes.fields.at(0)?.isPermeate || false;
+    return <div className=" text-red-600 bg-slate-300 border border-red-700 ">
+        <div className=" text-red-600 bg-slate-300">operational Times</div>
+        <div className="flex gap-2">
+            <input type="text" placeholder="command name" {...form.register('commandName')} />
+            <input type="text" placeholder="required entity" {...form.register('requiredEntity')} />
+        </div>
+        <div className="flex flex-col gap-2 pt-2">
+            {operationalTimes.fields.map((field, index) => <div key={field.id} className="flex gap-4 items-center">
+                <div>
+                    start time
+                    <input type="date" {...form.register(`operationalTimes.${index}.start.date`)} />
+                    <input type="time" {...form.register(`operationalTimes.${index}.start.time`)} />
+
+                </div>
+                <div>
+                    end time
+                    <input type="date" {...form.register(`operationalTimes.${index}.end.date`)} />
+                    <input type="time" {...form.register(`operationalTimes.${index}.end.time`)} />
+                </div>
+                {/* TODO: fix the colors */}
+                {index === 0 ? <>
+                    <button type="button" className={`${isPermeate ? ' text-green-700' : 'text-red-600'}`} onClick={() => {
+                    
+                        if (!isPermeate) form.setValue('operationalTimes', [{
+                            ...form.getValues('operationalTimes')[0],
+                            isPermeate: true
+                        }])
+                        else form.setValue(`operationalTimes.${index}.isPermeate`, !isPermeate)
+                        // operationalTimes.update(index, { isPermeate: !isPermeate })
+                    }}>
+                        permeate
+                    </button>
+                </> : <></>}
+                <button type="button"
+                    onClick={() => operationalTimes.remove(index)}>
+                    Remove Time
+                </button>
+            </div>)}
+        </div>
+        <div className="pb-2">
+            {/* FIXME: ask yinon and itamar about the time*/}
+            <button type="button" onClick={() => operationalTimes.append({
+                start: {
+                    date: moment().format(DATE_FORMAT),
+                    time: moment().format(TIME_FORMAT)
+                },
+                end: {
+                    date: moment().add(5, 'hours').format(DATE_FORMAT),
+                    time: moment().add(5, 'hours').format(TIME_FORMAT)
+                },
+                isPermeate: false
+            })}>Add Time</button>
+        </div>
+
+        <button type="button" onClick={() => props.nextStep()}>Next command</button>
+
+    </div>
+
+}
+
+export function CategoryDetails() {
```

Comment:
after the return 
1.please do write any more function only before the return 
2. you can write this function in better way
you can use here useFormContext for better readable code